### PR TITLE
Complete valid_to timestamps for entity versions missing from complete database imports

### PR DESCRIPTION
**Pull request by stevenleggdfe**
- Set valid_to timestamp for entity_versions when the ID didn't appear in a complete database import after the valid_from timestamp for that version
- Add logic to support sorting by ID in checksum calculation in addition to updated_at or created_at as appropriate
- Add logic to truncate timestamps to milliseconds in checksum calculations to ensure matching algorithms between dfe-analytics and dfe-analytics-dataform
- Add marts for imported and scheduled entity table checks - these enable checksum matches to be monitored proactively even when the assertions that monitor them automatically are disabled because ```compareChecksums: true``` is not set.
- Document data load reliability features in one place

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,6 +162,17 @@ If a field used to be included in streamed entity event data for a particular ta
 
 This may be useful if a field has been deleted from a table in the database, but you wish to continue to analyse past versions of that table which do contain that field, or changes between past versions of that table. The field will contain a ```NULL``` value for the latest version of the table.
 
+### Monitoring and correcting incomplete entity data received from dfe-analytics
+On rare occasions an entity in the application database may be created, updated or deleted, but no event streamed to your events table. This will result in missing or out of date data in tables generated by dfe-analytics.
+
+To detect incomplete or out of date tables, follow dfe-analytics documentation to enable nightly checksums to be streamed to your events table and set the ```compareChecksums: true``` parameter when you call ```dfeAnalyticsDataform()```. If the entity event stream in BigQuery appears to be missing, then the checksum in the event will not match the checksum calculated from the entity event stream in BigQuery. This will cause the ```foo_entity_ids_do_not_match``` assertion generated by dfe-analytics-dataform to fail.
```
Comment (ericaporter) [inline README.md:168]: Could possible say where the parameters are located

File: definitions/bat_apply_test.js [hunk 2]
```diff
@@ -3,12 +3,13 @@
 const dfeAnalyticsDataform = require("../");
 
 dfeAnalyticsDataform({
+    disabled: true,
```
Comment (ericaporter) [inline definitions/bat_apply_test.js:6]: What is this doing?

File: definitions/bat_register_test.js [hunk 3]
```diff
@@ -0,0 +1,966 @@
+/* For use by dfe-analytics-dataform developers working with Register test data only - for example code to use in your project, see definitions/example.js */
+
+const dfeAnalyticsDataform = require("../");
+
```
Comment (ericaporter) [inline definitions/bat_register_test.js:4]: I'm not sure what is happening in this file - is this to provide some test data for missing deletions work?

**Comment by ericaporter** [comment]
It all looks good - a few queries only

File: definitions/bat_apply_test.js (hunk 2 above)
Comment (stevenleggdfe) [reply definitions/bat_apply_test.js:6]: This is a set of Apply test data that I normally use for testing dfe-analytics-dataform. I switched it off with this parameter to make it faster to spot problems in the compiled SQL as well as less expensive to do test pipeline runs.

File: README.md (hunk 1 above)
Comment (stevenleggdfe) [reply README.md]: I've specified now that they go in the JSON passed to the function

**Review by ericaporter — APPROVED** [review]
LGTM 👍

**Pull request merged**